### PR TITLE
SNOW-886368: UDAF External Access Integration support

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-886368

- add external access integration to UDAF
- refactor test to dynamically create resource each time

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Review by sfc-gh-aalam — APPROVED** [review]

File: tests/integ/test_udaf.py [hunk 1]
```diff
@@ -498,3 +498,52 @@ def finish(self):
...
+            return self._sum
+
+        def accumulate(self, input_value):
+            import _snowflake
+            import requests
+
+            token = _snowflake.get_generic_secret_string("cred")
+            if (
+                token == "replace-with-your-api-key"
+                and requests.get("https://www.google.com").status_code == 200
```
Comment (sfc-gh-jrose) [inline tests/integ/test_udaf.py:520]: Should we hit our own resources instead of googles?

File: tests/integ/conftest.py [hunk 2]
```diff
@@ -47,6 +48,89 @@ def print_help() -> None:
...
+      MODE = EGRESS
+      TYPE = HOST_PORT
+      VALUE_LIST = ('www.microsoft.com');
+    """
+        ).collect()
+        session.sql(
+            f"""
+    CREATE OR REPLACE SECRET {key1}
+      TYPE = GENERIC_STRING
+      SECRET_STRING = 'replace-with-your-api-key';
```
Comment (sfc-gh-jrose) [inline tests/integ/conftest.py:79]: Are these strings passed in plaintext in the queries?

Comment (sfc-gh-aling) [reply tests/integ/conftest.py:79]: yes, they are passed as plaintext, it's for testing setup so we should be fine.

File: tests/integ/test_udaf.py (hunk 1 above)
Comment (sfc-gh-aling) [reply tests/integ/test_udaf.py:520]: I remember I tested snowflake before but I failed somehow..
on the other hand, the feature of external access integration is mostly for using external resources

**Review by sfc-gh-jrose — APPROVED** [review]

**Pull request merged**